File: Data Structures & Algorithms/max-water-container/submission-1.cpp
class Solution {
public:
    int maxArea(vector<int>& heights) {
        int max_area = 0;
        int start = 0;
        int end = heights.size() - 1;

        while (start <= end) {
            //calculate the area with every iteration
            max_area = max(max_area, area(heights[start], heights[end], end - start));
            
            if (heights[start] < heights[end]) {
                start++;
            }
            else if (heights[start] > heights[end]) {
                end--;
            }
            else {
                start++;
                end--;
            }
            
        }
        return max_area;
        
    }

    int area (int height1, int height2, int width) {
        return min(height1, height2) * width;
    }
};

File: Data Structures & Algorithms/buy-and-sell-crypto/submission-1.py
class Solution:
    def maxProfit(self, prices: List[int]) -> int:
        min_so_far = 101
        max_profit = 0
        for i in range(1, len(prices)):
            min_so_far = min(min_so_far, prices[i - 1])
            max_profit = max(max_profit, prices[i] - min_so_far)
        
        return max(0, max_profit)


        